File: API/Entities/Product.cs
﻿namespace API.Entities;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = default!;
    public string Description { get; set; } = default!;
    public long Price { get; set; }
    public string PictureUrl { get; set; } = default!;
    public string Type { get; set; } = default!;
    public string Brand { get; set; } = default!;
    public int QuantityInStock { get; set; }
}

File: UI/web/src/app/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  type: string;
  brand: string;
  quantityInStock: number;
}

export interface PaginatedProducts {
  products: Product[];
  metaData: ProductPagination;
}

export interface ProductOptions {
  types: string[];
  brands: string[];
}

export interface ProductFilters {
  orderBy: string;
  searchTerm?: string;
  types: string[];
  brands: string[];
  currentPage: number;
  pageSize: number;
}

export interface ProductPagination {
  currentPage: number;
  pageSize: number;
  totalPages: number;
  totalCount: number;
}
